src/main/java/pro/trevor/tankgame/rule/impl/shared/rule/PlayerRules.java

```diff
@@ -209,8 +209,19 @@ public static <T extends GenericTank> PlayerActionRule<T> SpendActionToShootWith
                 }
                 case BasicWall wall -> wall.setDurability(wall.getDurability() - 1);
                 case EmptyUnit emptyUnit -> {
-                    /* MISS */ }
-                default -> throw new Error("Unhandled tank shot onto " + u.getClass().getName());
+
+                }
+                case DestructibleFloor floor -> {
```

Comment:
You should test shooting a destructible floor with a unit on it.

Reply:
ACK. The expected behaviour is that the unit takes damage first while a unit is there.